app/models/photo.rb

```diff
@@ -0,0 +1,12 @@
+class Photo < ActiveRecord::Base
+  include AsUserAssociationExtension
+
+  belongs_to :user
+  has_many :comments, -> { extending WithUserAssociationExtension }, dependent: :destroy
+  has_many :likes, -> { extending WithUserAssociationExtension }, dependent: :destroy
+  has_attached_file :image, styles: { medium: "300x300>", large: "800x600>" }
+  validates_attachment_content_type :image, :content_type => /\Aimage\/.*\Z/
+
+  validates_presence_of :user
```

Comment:
Prefer the new style validations `validates :column, presence: value` over `validates_presence_of`.